### PR TITLE
James 2538 get/setFilter should accept null for ifInState and accountId

### PR DESCRIPTION
**Pull request by rouazana**

### CODE REVIEW

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/model/SetFilterRequest.java [hunk 1]
```diff
@@ -42,11 +42,17 @@ private Builder() {
         }
 
         public Builder accountId(String accountId) {
-            throw new JmapFieldNotSupportedException(ISSUER, "accountId");
+            if (accountId != null) {
```
Comment (blackheaven) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/model/SetFilterRequest.java]: If I remember well, we are supposed to use `Optional.orElseThrow` in this case.

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/model/SetFilterRequest.java [hunk 2]
```diff
@@ -42,11 +42,17 @@ private Builder() {
...
+                throw new JmapFieldNotSupportedException(ISSUER, "accountId");
+            }
+            return this;
         }
 
         public Builder ifInState(String ifInState) {
-            throw new JmapFieldNotSupportedException(ISSUER, "ifInState");
+            if (ifInState != null) {
```
Comment (blackheaven) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/model/SetFilterRequest.java]: Idem

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/model/SetFilterRequest.java (hunk 1 above)
Comment (rouazana) [reply server/protocols/jmap/src/main/java/org/apache/james/jmap/model/SetFilterRequest.java]: I don't see the interest while you don't need to get/instantiate the value

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/model/SetFilterRequest.java (hunk 2 above)
Comment (rouazana) [reply server/protocols/jmap/src/main/java/org/apache/james/jmap/model/SetFilterRequest.java]: idem :)

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/FilterTest.java [hunk 3]
```diff
@@ -79,6 +79,7 @@ public void setup() throws Throwable {
         RestAssured.requestSpecification = jmapRequestSpecBuilder
             .setPort(jmapServer.getProbe(JmapGuiceProbe.class).getJmapPort())
             .build();
+        RestAssured.enableLoggingOfRequestAndResponseIfValidationFails();
```
Comment (aduprat) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/FilterTest.java]: Not really found of this has it leads to a lot of logs when calling _await_ method.

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/model/SetFilterRequest.java (hunk 1 above)
Comment (chibenwa) [reply server/protocols/jmap/src/main/java/org/apache/james/jmap/model/SetFilterRequest.java]: It's more

```
Optionnal.ofNullable(X).ifPresent(any -> throw new JmapFieldNotSupportedException(ISSUER, "accountId"));
```

But @rouazana  version is fine to me

**Comment by aduprat** [comment]
merged

**Pull request closed without merging**